src/components/table/JadeTable.jsx

```diff
@@ -4,98 +4,184 @@ import { withStyles } from '@material-ui/core/styles';
+    const offset = page * rowsPerPage;
+    this.setState({ page }); // offset
+    handleFilter(rowsPerPage, offset, orderBy, orderDirection, searchString);
+  };
 
-    this.setState({ order: newOrder, orderBy: property });
+  handleSearchString = event => {
+    const { handleFilter } = this.props;
+    const { page, orderDirection, orderBy, rowsPerPage } = this.state; // limit
+    const offset = page * rowsPerPage;
+    const searchString = event.target.value;
+    this.setState({ searchString }); // filter
+    handleFilter(rowsPerPage, offset, orderBy, orderDirection, searchString);
   };
 
   render() {
-    const { classes, columns, rows } = this.props;
-    const { order, orderBy } = this.state;
+    const { classes, columns, rows, summary, totalCount } = this.props;
+    const { orderBy, orderDirection, page, rowsPerPage } = this.state;
+    const emptyRows = rowsPerPage - Math.min(rowsPerPage, 8 - page * rowsPerPage);
     return (
-      <Paper className={classes.root}>
-        <Table className={classes.table}>
-          <JadeTableHead
-            columns={columns}
-            onRequestSort={this.handleRequestSort}
-            order={order}
-            orderBy={orderBy}
-          />
-          <TableBody>
-            {stableSort(rows, getSorting(order, orderBy)).map(row => (
-              <TableRow key={row.id} className={classes.row}>
-                {columns.map(col => (
-                  <TableCell key={col.property}>
-                    {col.render ? col.render(row) : row[col.property]}
-                  </TableCell>
-                ))}
-              </TableRow>
-            ))}
-          </TableBody>
-        </Table>
-      </Paper>
+      <div>
+        {!summary && (
+          <div className={classes.search}>
+            <div className={classes.searchIcon}>
+              <SearchIcon />
+            </div>
+            <InputBase
+              placeholder="Search"
+              classes={{
+                root: classes.inputRoot,
+                input: classes.searchInput,
+              }}
+              onChange={this.handleSearchString}
+            />
+          </div>
+        )}
+        <Paper className={classes.root}>
+          <Table className={classes.table}>
+            <JadeTableHead
+              columns={columns}
+              onRequestSort={this.handleRequestSort}
+              order={orderDirection}
+              orderBy={orderBy}
+            />
+            <TableBody>
+              {rows.map(row => (
+                <TableRow key={row.id} className={classes.row}>
+                  {columns.map(col => (
+                    <TableCell key={col.property}>
+                      {col.render ? col.render(row) : row[col.property]}
+                    </TableCell>
+                  ))}
+                </TableRow>
+              ))}
+              {rows.length < rowsPerPage && (
+                <TableRow style={{ height: 50 * emptyRows }}>
```

Comment:
magic number

Reply:
do I want to pass the styling theme into the react component?

Reply:
I don't think so unless we need it inside. I was just thinking of defining this as `ROW_HEIGHT` or something up above.